### DISCUSSION TITLE
SPDIF out interface

### DISCUSSION THREAD

**Discussion by phe78** [probonopd/MiniDexed#218] [General]
Hi folks,
The TX816 has been a Graal for me when I was a teenager… imagine how I reacted when I realized I could get the equivalent with a raspi and other stuff for cheap … WARM thanks to you guys who managed to not only develop this but also make it approachable with clear instructions for me to make it work !!
Now my question : my studio is running out of analog inputs but still has SPDIF coax and optical ins available. Hence I’d like to add SPDIF outs to my raspberry. I saw some I2S boards such as this one : https://www.amazon.fr/dp/B07DK5C4TY?ref_=cm_sw_r_cp_ud_dp_WDVNDFCBMYAQQSEFPQ6X
Would that work with MiniDexed ? Also' would that work if connected with a PCM5122 analog out board at the same time ?
many thanks in advance for your feedback
Philippe

**Comment by probonopd** [comment]
Welcome @phe78. I don't think you can attach more than one i2s device at a time.

That board uses a WM8804G DAC. We are not listing this one as compatible at the moment. I doubt that S/PDIF would work without software changes anyway, because as I understand the data for S/PDIF is not generated from the i2s signal which we are using (not 100% sure on this one). So no, I don't think this will work currently. For professional use S/PDIF should probably be avoided due to jitter anyway and i2s should be preferred.

**Comment by sou7611** [comment]
Can a USB audio output be driven?  If so, that might be an access for a
digital audio signal (which could then be converted to AES or SPDIF)

**Reply by probonopd** [reply]
Don't professionals avoid S/PDIF?

**Reply by luisfcorreia** [reply]
not really, there are digital tribes

I for example, use ADAT to connect my two audio cards :)

**Comment by luisfcorreia** [comment]
USB Audio is not supported
(at the moment, at least)

**Comment by sou7611** [comment]
Well, then I'd look at it this way : none of the Yamaha gear supported
digital outputs.  Heck, not all of 'em even supported balanced outputs.

**Comment by sou7611** [comment]
SPDIF is fine as long as you have a free running clock, it's essentially
the same thing as stereo AES.

Where people get in trouble is when you need synced clocks, because SPDIF
can't do that (but AES can)

I've seen and used both formats, and many digital devices (mixers,
interfaces, recorders, etc) can interface with both as well.

SPDIF does have the advantage of optical connections, if you want to go
that route.

**Reply by sou7611** [reply]
Something else to consider as a way to extract a digital audio stream... Use a HDMI audio extractor to extract the digital audio, e.g. an HDMI audio de embedder.

There are several inexpensive models on Amazon that output SPDIF in addition to analog audio.

**Reply by probonopd** [reply]
Yes, that should work.

**Reply by phe78** [reply]
Many thanks to you all for your great replies ! I didn't think about the HDMI audio extractor, thats a great idea. So for the time being I'll go with my 5122 DAC board. I just made a test with breadboard connections, worked great right away. Now build a board with plugs connecting the display encoder and GPIOs. Will keep you folks posted !
